tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/ManualInstrumentation/Tracer/ConfigureIntegration.cs

```diff
@@ -180,7 +180,7 @@ internal static void UpdateSettings(Dictionary<string, object?> dictionary, Trac
                     break;
 
                 case TracerSettingKeyConstants.ServiceNameMappingsKey:
-                    if (setting.Value is Dictionary<string, string> mappings)
+                    if (setting.Value is Dictionary<string, string?> mappings)
```

Comment:
I don't think we should allow null values - though i'm not sure if we should filter them out in the manual instrumentation side, here, or in the SetServiceNameMappings function